### PR TITLE
Add enhanced-sampling submodule: metadynamics, boxed dynamics, and loxodynamics

### PR DESCRIPTION
**Pull request by amateurcat**
Adds a self-contained torch_sim.enhanced_sampling submodule of enhanced-sampling methods that compose with any MLIP via SumModel

### CODE REVIEW

File: torch_sim/enhanced_sampling/metadynamics.py [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+    Returns:
+        Per-system sums where the atom dimension is replaced by n_systems.
+    """
+    out_shape = list(src.shape)
+    out_shape[dim] = n_systems
+    out = torch.zeros(out_shape, device=src.device, dtype=src.dtype)
+    return out.index_add(dim, system_idx, src)
+
+
+class LogfermiWall(ModelInterface):
```
Comment (CompRhys) [inline torch_sim/enhanced_sampling/metadynamics.py:70]: Can we append `Model` to the name here to name it clearer?

File: torch_sim/enhanced_sampling/metadynamics.py [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+        energy = _segment_sum(v_atom, system_idx, state.n_systems)
+
+        # F = -dV/dr * r_hat with dV/dr = k_wall * beta * sigmoid(x)
+        dv_dr = self.k_wall * self.beta * torch.sigmoid(x)
+        forces = -dv_dr.unsqueeze(-1) * dvec / r.unsqueeze(-1)
+
+        return {"energy": energy, "forces": forces, self.energy_label: energy}
+
+
+class RMSDCV(ModelInterface):
```
Comment (CompRhys) [inline torch_sim/enhanced_sampling/metadynamics.py:168]: Can we append `Model` to the name here to name it clearer?

File: torch_sim/enhanced_sampling/skewencoder.py [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+        """Validate the configuration."""
+        if self.latent_dim != 1:
+            raise ValueError(f"{self.latent_dim=} must be 1 in this version")
+        if self.input_dim < 1:
+            raise ValueError(f"{self.input_dim=} must be >= 1")
+        if self.activation not in _ACTIVATIONS:
+            raise ValueError(f"{self.activation=} must be one of {_ACTIVATIONS}")
+
+
+class ShiftedSoftplus(nn.Module):
```
Comment (CompRhys) [inline torch_sim/enhanced_sampling/skewencoder.py:79]: is there no common library that already defines this?

File: torch_sim/enhanced_sampling/skewencoder.py [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+        "loss_reconstruction": loss_ae.detach(),
+        "loss_skew": loss_skew.detach(),
+        "loss_l2": (beta * l2).detach(),
+        "local_skewness": skewness_1d(z_local, eps=eps).detach(),
+    }
+    return loss_total, diagnostics
+
+
+@dataclass
+class DescriptorNormalizer:
```
Comment (CompRhys) [inline torch_sim/enhanced_sampling/skewencoder.py:252]: again do we need to implement this here rather than import a standard mean-variance normalizer?

**Comment by CompRhys** [comment]
Thanks for the feature contribution! I gave it a quick glance today but will look in more details next week. Comments thus far are just nits not looking at the algorithms/test coverage yet.
